.github/workflows/release-binaries.yml

```
@@ -0,0 +1,30 @@
+# This workflow is meant to trigger a build of Docker binaries when a tag
+# is created, it uses the existing `build-docker-image.yml` workflow
+#
+# We use a separate action as we might want to trigger this with different under
+# different circumstances than a Continous Deployment, for example.
+name: Release binaries
+
+on:
+  create:
```

I'm not sure if we want to use tag filters.

We can avoid them by building the image when a pre-release is "published".
(Pre-releases don't appear on the Zebra front page, and they have a pre-release tag in the release list.)

Then when the image is ready, we can change the pre-release to a release (the "released" event).

Here are the details:
https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-object-41

```suggestion
on:
  release:
    types: [published]
```

---

I decided on a slightly different approach

The event type `published` would also publish images to DockerHub when creating a pre-release, but we can keep those in GAR (Google Artifact Registry), if we'd like to publish it as an official release (even if that's a `beta`) then we would have to _un-check_ it as a `pre-release` in GitHub, and it would also be published to DockerHub.

This is mainly to keep DockerHub with just releases meant to be used in production, or after the images has been **tested** (as we should test those before deploying the binaries to DockerHub)

---

As expected.

When a release is published as `pre-release` it triggers our CD for internal testing: https://github.com/ZcashFoundation/zebra/actions/runs/3048068297

When the `pre-release` is _unchecked_ (an official release) then it only runs the `Release binaries` workflow https://github.com/ZcashFoundation/zebra/actions/runs/3048093839

We should wait for both of these tests to finish